### PR TITLE
Update Gemma2 attention scale

### LINKED ISSUE #693: [Bug Report] Gemma-2-2b-it output logit doesn't match with huggingface
**Describe the bug**
The output logits from transformer_lens and huggingface are quite different using Gemma-2-2b-it model

**Code example**
```python
import torch
import transformer_lens
from transformers import AutoTokenizer, AutoModelForCausalLM

device = 'cuda'
model_name = 'google/gemma-2-2b-it'
tl_model = transformer_lens.HookedTransformer.from_pretrained(model_name, device=device)

tokenizer = AutoTokenizer.from_pretrained(model_name)
hf_model = AutoModelForCausalLM.from_pretrained(model_name).to(device)

inputs = tokenizer('Hello world', return_tensors="pt").to(device)

logits_tl = tl_model(inputs.input_ids, return_type='logits', prepend_bos=False)
logits_hf = hf_model(**inputs).logits

print((logits_tl[0, -1] - logits_hf[0, -1]).mean()) # 0.1159
print((logits_hf[0, -1]).min(), (logits_hf[0, -1]).max()) # -19.6916 16.0789
```

**System Info**
transformer_lens 2.3.0, transformers 4.43.2

**Additional context**
The logit diff is quite large

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/TransformerLensOrg/TransformerLens/issues) in the repo (**required**)

### PR DESCRIPTION
**Pull request by mntss**
# Description

Current configuration uses incorrect attention scale. According to deepmind implementation the 2b and 9b versions use sqrt(d_head). This is the default scale in TL.
https://github.com/google-deepmind/gemma/blob/a0504162f99a1c238efb37b8197e711c0f3808fd/gemma/transformer.py#L152-L174

Fixes #693 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

### Screenshots
Please attach before and after screenshots of the change if applicable.

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### CODE REVIEW

**Comment by neelnanda-io** [comment]
Huh, you seem to be correct my bad. I'm not sure why my sanity checks didn't show an enormous divergence
Can you check if the attention patterns are notably closer to HuggingFace after this change?

**Comment by mntss** [comment]
HF implementation uses SPDA so I'd need to switch the attention implementation to access the pattern. I compared the output instead (`hook_z`) across layers

image

**Comment by ccp123456** [comment]
This question is very helpful for me and I am also being confused by this question! How do I adjust the Transformerlens' Gemma to ensure that the Transformerlens Gemma gives the same results as the HF? Which file should I add the above code to?

**Comment by bryce13950** [comment]
@mntss Thank you very much for this! @ccp123456 this will be put into a release relatively quickly, so you should be able to make use of this right away. The question of 100% accuracy is a little bit more complicated, and making sure Gemma models are 100% accurate is a pretty high priority task at the moment. If you interested in knowing more, DM me on slack. (If you are not on the slack channel, let me know and I can give you access.)
